syntax = "proto3";

package mach.proto;
option go_package = "github.com/ryepropgroup/protoServer/protos/messages";

message SensorData {
    int64 timestamp = 1;
    map<string, float> values = 2;
}

message DeviceInformation {
    map<string, Device> devices = 1;
    repeated string sequences = 2;
}

message SequenceCommand {
    string sequence = 1;
}

message Device {
    enum DeviceType {
        SENSOR = 0;
        VALVE = 1;
    }
    enum DeviceLocation {
        GSE = 0;
        REMOTE = 1;
    }
    int32 pin = 1;
    DeviceType type = 2;
    DeviceLocation location = 3;
}

message Response {
    bool success = 1;
    string message = 2;
}
